rootfs/scheduler/__init__.py

```diff
@@ -1082,17 +1102,33 @@ def _buildpack_readiness_probe(self, controller, delay=30, timeout=5, period_sec
                 'failureThreshold': failure_threshold,
             },
         }
+        return readinessprobe
 
-        # Update only the application container with the health check
-        app_type = controller['spec']['selector']['type']
-        namespace = controller['spec']['selector']['app']
-        container_name = '{}-{}'.format(namespace, app_type)
-        containers = controller['spec']['template']['spec']['containers']
-        for container in containers:
-            if container['name'] == container_name:
-                container.update(readinessprobe)
-
-        return controller
+    '''
+    Applies tcp socket readiness probe to the docker app container only if some port is exposed
+    by the docker image.
+    '''
+    def _default_dockerapp_readiness_probe(self, image, delay=5, timeout=5, period_seconds=5,
+                                           success_threshold=1, failure_threshold=1):
+        port = self._get_port(image)
+        if port is None:
+            return None
```

Comment:
if the docker app's service is defaulting to port 5000, why aren't we doing that here? Is there a case where we don't want to add a readiness check here? If so let's document this in the code so others can understand the reasoning behind this.